### PR TITLE
MAINT: remove non-default settings (except `shallow`) in `.gitmodules`

### LINKED ISSUE #15276: Boost and PROPACK git submodules are too easy to commit changes for
I've now been bitten about 5 times by the latest change in the PROPACK submodule, either committing it by accident myself or seeing that in other PRs. The problem is the combination of `active` and `ignore` settings in `.gitmodules`:

```
[submodule "doc/sphinxext"]
	path = doc/sphinxext
	url = https://github.com/numpy/numpydoc.git
[submodule "doc/source/_static/scipy-mathjax"]
	path = doc/source/_static/scipy-mathjax
	url = https://github.com/scipy/scipy-mathjax.git
[submodule "scipy/_lib/boost"]
	path = scipy/_lib/boost
	url = https://github.com/scipy/boost-headers-only
	shallow = true
	active = false
	ignore = all
[submodule "scipy/sparse/linalg/_propack/PROPACK"]
	path = scipy/sparse/linalg/_propack/PROPACK
	url = https://github.com/scipy/PROPACK
	shallow = true
	active = false
	ignore = all
[submodule "scipy/_lib/unuran"]
	path = scipy/_lib/unuran
	url = https://github.com/scipy/unuran.git
```

This makes changes to those submodules invisible, for example in the output of `git status` and `gitk`. So when you check what you're committing with `git status`, you can't see you're about to commit a change of the commit that the PROPACK submodule points to. And when you're switching branches a lot, such changes happen all the time. Normally you rely on `git status` to catch that, and then run `git submodule update` to fix it.

@mckib2 is there any reason why you added those settings? Can we simply remove them?

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-15276

[ci skip]

### CODE REVIEW

**Review by mckib2 — APPROVED** [review]

**Comment by tylerjereddy** [comment]
I've added a backport label--otherwise, paranoia about silent submodule updates, or failure to update, is going to cost time on the `maintenance/1.8.x` branch, where some future PROPACK changes for Windows cannot be ruled out yet.

**Comment by rgommers** [comment]
good idea
